### PR TITLE
Add Sparra/HHG code to pipeline in `run_episode_file` 

### PR DESCRIPTION
**Pull request by Jennit07**
Opening this as a draft PR for now as this needs testing against an episode file. However i have written the code to match on the SPARRA and HHG variables and included this in the processing for the episode file

### CODE REVIEW

File: R/run_episode_file.R
Comment (Moohan) [inline R/run_episode_file.R:1]: We usually do this: e.g. for 2019/20 `sparra/hhg-1920 = sparra/hhg_start_fy`  `sparra/hhg-2021 = sparra/hhg_end_fy`

So we end up with 4 variables for most years: `hhg_start_fy`, `sparra_start_fy`, `hhg_end_fy` and `sparra_end_fy`. And just 2 variables, `*_start_fy` or `*_end_fy` for the first years, or the most recent years.

Comment (Jennit07) [reply R/run_episode_file.R:1]: Changes made, ready for review

File: R/check_year_valid.R [hunk 1]
```diff
@@ -24,12 +24,17 @@ check_year_valid <- function(year, type = c(
...
+                               "SDS",
+                               "Sparra-HHG"
                              )) {
   if (year >= "2223" & type == "NSU") {
     return(FALSE)
   }
 
+  if (year >= "2223" & type == "Sparra-HHG") {
+    return(FALSE)
+  }
```
Comment (Moohan) [inline R/check_year_valid.R:36]: SPARRA and HHG have different year limits.

SPARRA starts for 1516 (so end of year 1415), and HHG starts in 1819 (so end of year 1718). HHG was discontinued (#576) so 2223 is the last year we'll get (so 2223 will only ever have start of year, and subsequent years will have none).

File: R/run_episode_file.R [hunk 2]
```diff
@@ -326,3 +326,76 @@ join_cohort_lookups <- function(ep_file_data, year) {
...
+        by = "chi"
+      ) %>%
+      dplyr::full_join(
+        read_file(get_hhg_path(fy)) %>%
+          dplyr::rename(
+            chi = "upi_number",
+            hhg_start_fy = "hhg_score"
+          ),
+        by = "chi"
+      )
```
Comment (Moohan) [inline R/run_episode_file.R:360]: Can you make the next_fy bit into a function, please?
```suggestion
  fy <- as.integer(substr(year, 1, 2))
  next_fy <- paste0(fy + 1L, fy + 2L)

  if (!check_year_valid(year, "Sparra-HHG")) {
    join_sparra_hhg_file <- ep_file_data %>%
      dplyr::full_join(
        read_file(get_sparra_path(year)) %>%
          dplyr::rename(
            chi = "upi_number",
            sparra_start_fy = "sparra_risk_score"
          ),
        by = "chi"
      ) %>%
      dplyr::full_join(
        read_file(get_hhg_path(year)) %>%
          dplyr::rename(
            chi = "upi_number",
            hhg_start_fy = "hhg_score"
          ),
        by = "chi"
      )
```

File: R/run_episode_file.R [hunk 3]
```diff
@@ -326,3 +326,76 @@ join_cohort_lookups <- function(ep_file_data, year) {
...
+    dplyr::full_join(
+      read_file(get_hhg_path(next_fy)) %>%
+        dplyr::rename(
+          chi = "upi_number",
+          hhg_end_fy = "hhg_score"
+        ),
+      by = "chi"
+    )
+
+  return(join_sparra_hhg_file)
```
Comment (Moohan) [inline R/run_episode_file.R:400]: I think this is obsolete?
```suggestion
```

File: R/run_episode_file.R [hunk 4]
```diff
@@ -326,3 +326,65 @@ join_cohort_lookups <- function(ep_file_data, year) {
 
   return(join_cohort_lookups)
 }
+
+#' Join SPARRA and HHG
+#'
+#' @inheritParams store_ep_file_vars
+#'
+#' @return The data including the SPARRA and HHG variables matched
+#' on to the episode file.
```
Comment (Moohan) [inline R/run_episode_file.R:335]: This should be in it's own script as it will also be used in the indiv file

Comment (Jennit07) [reply R/run_episode_file.R]: Done

**Review by Moohan — APPROVED** [review]
I think this is good to merge now

**Pull request merged**